src/shogun/structure/MAPInference.cpp

```diff
@@ -38,6 +39,9 @@ CMAPInference::CMAPInference(CFactorGraph* fg, EMAPInferType inference_method)
 		case GRAPH_CUT:
```

Comment:
Definitely a good idea! Some of functions in CFactorType CFactor for operating energy tables are hard to follow. You may open an issue for this. So either you or I can do it when we have time.